bank_atm.rb

```diff
@@ -0,0 +1,96 @@
+$LOAD_PATH << '.'
+require 'validations'
+require 'models'
+
+#Begin of Demo data
+p1 = Person.new("John", "Developer", "j@yahoo.com")
+p2 = Person.new("Tudor", "Climbing instructor", "t@tahoo.com")
+
+persons = []
+persons << p1
+persons << p2
+
+b1 = BankAccount.new("1234", p1, 150)
+b2 = BankAccount.new("5555", p1, 5500)# with 1 widthrawal already
+b3 = BankAccount.new("9999", p2, 7000)
+
+b2.withdraw(4000)
+#End of demo data
+
+#Start of the flow
+loop do
+    puts "\nWelcome to the ATM!"
+    puts "Select an option: "
+    #Options
+    puts "1. Deposit funds"
+    puts "2. Withdraw funds"
+    puts "3. Create account"
+    puts "4. View all accounts"
+    puts "5. Quit"
+
+    option = gets.chomp.downcase
+    break if option == "5"
+    puts "You selected #{option}"
+
+    case option
+        when "1" #deposit
+            puts "Insert your name:"
+            input_name = gets.chomp
+            puts "Insert PIN:"
+            input_pin = gets.chomp
+
+            #verify pin
+            current_user = persons.find {|person| person.name == input_name}
+            if current_user == nil 
+                puts "No user found with that name"
+
+            #find the bank account
+            elsif bank_acc = current_user.bank_accounts.find {|account| account.pin_matches?(input_pin) }
```

Comment:
I see this is a repeated part of code, you should create a method with that and reuse it